Document/0x05d-Testing-Data-Storage.md

```diff
@@ -444,34 +442,32 @@ Android provides users with an auto-backup feature. The backups usually include
 
 - OEMs may provide additional options. For example, HTC devices have a "HTC Backup" option that performs daily backups to the cloud when activated.
 
-Apps must be careful to ensure that sensitive user data stored by the application is not placed within these backups as this may allow an attacker to extract app data from the backup if obtained.
+Apps must carefully ensure that sensitive user data doesn't end within these backups as this may allow an attacker to extract it.
 
 ### Process Memory
 
-All applications on Android use memory to perform normal computational operations like any regular modern-day computer. It is of no surprise then that at times sensitive data and sensitive operations will be held and performed within process memory.
-
-For this reason it is important for applications to expose sensitive information as briefly as possible in memory. Once the relevant sensitive data has been used it should be disposed of from process memory.
+All applications on Android use memory to perform normal computational operations like any regular modern-day computer. It is of no surprise then that at times sensitive operations will be performed within process memory. For this reason, it is important that once the relevant sensitive data has been processed, it should be disposed from process memory as quickly as possible.
 
-The investigation of an application's memory can be done from memory dumps, and from analyzing the memory in real time via a debugger. Its important to note that the analyse of memory is quite error prone and that any approach to analyse of memory takes this into account.
+The investigation of an application's memory can be done from memory dumps, and from analyzing the memory in real time via a debugger. It's important to note that memory analysis is quite error prone and that any approach to analyze memory takes this into account.
```

Comment:
Unfortunately I still don't get this sentence completely, could you please try to explain what's meant by "any approach takes this into account"?

Reply:
This was taken from the original overview, I agree its not very specific.

Personally I understood it to mean that because memory is naturally dynamic the act of dumping and going through memory is prone to missing things as it all depends on when you dumped the memory. 

But honestly its so unspecific that maybe I'm even misunderstanding the original authors meaning. Once i action the other commits i'll come back and reword this to what I think it means and we cant take it from there.

Reply:
Actually thinking about it I dont think its important to mention that in the overview, its more of a test case specific detail.